app/docker/networks/edit/NetworkDetailsTable.tsx

```diff
@@ -0,0 +1,145 @@
+import { Fragment } from 'react';
+import DockerNetworkHelper from 'Docker/helpers/networkHelper';
+
+import { Widget, WidgetBody, WidgetTitle } from '@/portainer/components/widget';
+import {
+  DetailsTable,
+  DetailsTableKeyValueRow,
+} from '@/portainer/components/DetailsTable';
+import { Button } from '@/portainer/components/Button';
+import { Authorized } from '@/portainer/hooks/useUser';
+
+import { isSystemNetwork } from '../network.helper';
+import { DockerNetwork, IPConfig } from '../types';
+
+interface Props {
+  network: DockerNetwork;
+  onRemoveNetworkClicked: () => void;
+}
+
+type NetworkKey =
+  | 'Name'
+  | 'Id'
+  | 'Scope'
+  | 'Driver'
+  | 'Attachable'
+  | 'Internal';
+
+type NetworkRowContent = [NetworkKey, string][];
+
+const filteredNetworkKeys: NetworkKey[] = [
+  'Name',
+  'Id',
+  'Driver',
+  'Scope',
+  'Attachable',
+  'Internal',
+];
+
+export function NetworkDetailsTable({
+  network,
+  onRemoveNetworkClicked,
+}: Props) {
+  const networkRowContent: NetworkRowContent = filteredNetworkKeys.map(
+    (key) => [key, String(network[key])]
+  );
+  const allowRemoveNetwork = !isSystemNetwork(network.Name);
+  const IPV4Configs: IPConfig[] | undefined =
+    network.IPAM && DockerNetworkHelper.getIPV4Configs(network.IPAM.Config);
+  const IPV6Configs: IPConfig[] | undefined =
+    network.IPAM && DockerNetworkHelper.getIPV6Configs(network.IPAM.Config);
```

Comment:
will look cleaner:
```suggestion
  const ipv4Configs: IPConfig[] = DockerNetworkHelper.getIPV4Configs(network.IPAM?.Config || []);
  const ipv6Configs: IPConfig[] = DockerNetworkHelper.getIPV6Configs(network.IPAM?.Config | [])
```

this will always make sure ipv4Configs and ipv6Configs will always be an array and never undefined

Reply:
updated